### PR TITLE
Unconditionally enable WITH MUTUALLY RECURSIVE

### PR DESCRIPTION
**Pull request by umanwizard**
### Motivation

   * This PR refactors existing code.

WMR has been on for all production users for a while. This change simplifies the code and turns it on for `bin/environmentd` without needing extra flags.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - None.

### CODE REVIEW

File: test/sql-feature-flags/mzcompose.py [hunk 1]
```diff
@@ -226,35 +226,6 @@ def query_item(cls, ordinal: int) -> str:
...
-
-    @classmethod
-    def drop_item(cls, ordinal: int) -> str:
-        return f"DROP VIEW wmr_{ordinal:02d}"
-
-    @classmethod
-    def query_item(cls, ordinal: int) -> str:
-        return f"SELECT * FROM wmr_{ordinal:02d}"
-
-
```
Comment (antiguru) [inline test/sql-feature-flags/mzcompose.py:257]: If it doesn't exist already, we might want to change this into a platform check.

**Review by antiguru — APPROVED** [review]

**Comment by antiguru** [comment]
I'm surprised to see so little changes, which means that we don't have good test coverage for WMR. There should be an upgrade test and a platform check, but both seem to be missing. If you merge this, please file issues for QA to add such tests.

File: test/sql-feature-flags/mzcompose.py (hunk 1 above)
Comment (umanwizard) [reply test/sql-feature-flags/mzcompose.py:257]: Now that there isn't a feature flag, this isn't really testing anything other than that a simple WMR view (the `foo` above) can be created, queried, and dropped without things breaking. I am sure we already have test coverage for such a case.

**Comment by umanwizard** [comment]
It's possible that we do have good test coverage, but that this isn't changing anything because WMR is already enabled in CI. @philip-stoev any thoughts?

**Comment by aalexandrov** [comment]
> I'm surprised to see so little changes, which means that we don't have good test coverage for WMR.

Most `WITH MUTUALLY RECURSIVE` tests are in `*.slt` files which already run with the feature flag enabled, so they are not touched by this PR.

**Comment by aalexandrov** [comment]
There should be an item on the release that deploys this to also archive the feature flag in LaunchDarkly.

**Comment by philip-stoev** [comment]
In a separate PR, I am going to add a platform check for WITH MUTUALLY RECURSIVE, which will also serve as an upgrade test.  Thank you.

**Pull request merged**